### ISSUE TITLE
Cancellable Queries

### ISSUE THREAD

**Issue by bradleygore** [grncdr/node-any-db-pool#15]
Would be awesome if there was a way to cancel queries that are queued up in the pool. We're using postgres and, though there are [open issues](https://github.com/brianc/node-postgres/issues/773) with the cancel implementation in `node-postgres`, we can't even try the workarounds when pooling as the pooled queries don't seem to expose the client they are getting run with, etc...

Is there something we're missing - or is this just not possible?

**Comment by grncdr** [comment]
At the moment it's not possible. It seems reasonable to implement cancellation for queued queries, but I'm not sure about cancelling queries themselves once they've been passed off to a client. As not all clients will support cancellation.

For example, an API like `pool.cancel(query)` would make it more clear that the query itself is not necessarily cancellable. Would that work for your use case?

**Comment by bradleygore** [comment]
@grncdr Yeah, at least cancelling those that have not yet been passed off to a client would be a major help!

If there was a way to add a ref to the `client` once a queued query has been passed to one, that would also then allow consumers to use the `client` per whatever underlying adapter they are using. Something like this:

```
var hugeReportQueriesQueue = [];

// get list of all queries to run to get data for a big report, i.e. enterprise level or some such
// assume they are all structured appropriately to pass to pool.query
var allQueriesToRun = collectReportQueries();

// loop through, get the resulting Query object from pool.query and push onto list of hugeReportQueriesQueue
allQueriesToRun.forEach(query => hugeReportQueriesQueue.push(pool.query(query)));

// add listener for the request to be cancelled (i.e. in an express app or some such)
req.on('close', () => hugeReportQueriesQueue.forEach(query => {
  if (query.client) { 
    // then query was passed to client, can pull in node-postgres directly and interact with it here
    pg.cancel(query.client.connectionParameters, query.client, query);
  } else {
    // query still pooled, and not sent to db adapter yet - cancel from the pool
    pool.cancel(query);
  }
}));
```

**Comment by bradleygore** [comment]
Hey @grncdr I appreciate this getting merged. Unfortunately, current `any-db` specifies only older version of `any-db-pool`, so I created [this pr](https://github.com/grncdr/node-any-db/pull/74) so that it bumps version up a patch and then updates version of `any-db-pool` it depends on.

**Comment by bradleygore** [comment]
Hey @grncdr - I wanted to follow up and see if you received the comment I made about publishing to npm. The `any-db` codebase was updated, but it wasn't published to npm. So, I can't get the update. Would it be possible to run the `npm publish` to get this up there? Thanks so much again for adding in the cancellation feature to the pool - it will be just what we needed 😃
